### PR TITLE
Feature/xss escaping modules

### PR DESCRIPTION
**Pull request by joeyblake**
**Types of change**

- [ ] Bug fix (change which fixes an issue)
- [x] Enhancement (change which improves upon an existing feature)
- [ ] New feature (change which adds new functionality)

**Description**

Adds context specific late escaping filters to improve XSS security. This ensures that escaping does not interfere with other logic use cases, in the code it is clear that the string is escaped at output, and any changes further up in the logic will not open up the same vulnerabilities in the future.

https://developers.hubspot.com/docs/cms/hubl/filters#escape-html
https://developers.hubspot.com/docs/cms/hubl/filters#escape-attr
https://developers.hubspot.com/docs/cms/hubl/filters#escape-url
https://developers.hubspot.com/docs/cms/hubl/filters#escape-js
https://developers.hubspot.com/docs/cms/hubl/filters#sanitize-html

**Relevant links**

Example page:
GitHub issue:

**Checklist**

- [x] I have read the [CONTRIBUTING](https://github.com/HubSpot/cms-theme-boilerplate/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the [style guide requirements](https://github.com/HubSpot/cms-theme-boilerplate/blob/master/STYLEGUIDE.md).
- [x] I have thoroughly tested my change.

**People to notify**

### CODE REVIEW

File: src/modules/menu.module/module.html [hunk 1]
```diff
@@ -86,24 +86,33 @@
...
         {% if depth and depth < module.max_levels %}
+          {% set depth = depth + 1 %}
           <button class="menu__child-toggle no-button" aria-expanded="false">
-            <span class="show-for-sr">Show submenu for {{ link.label }}</span>
+            <span class="show-for-sr">Show submenu for {{ link.label|escape_html }}</span>
             <span class="menu__child-toggle-icon"></span>
           </button>
-          <ul class="menu__submenu menu__submenu--level-{{ depth + 1 }} no-list">
-            {% set depth = depth + 1 %}
+          <ul class="menu__submenu menu__submenu--level-{{ depth|escape_attr }} no-list">
```
Comment (jasonnrosa) [inline src/modules/menu.module/module.html:115]: The `+1` piece of the variable looks like it got dropped here

**Review by jasonnrosa — COMMENTED** [review]
Looking good to me! Just one comment on one of the variables used for menu depth

File: src/modules/menu.module/module.html (hunk 1 above)
Comment (joeyblake) [reply src/modules/menu.module/module.html:115]: @jasonnrosa previously we were doing this math twice once in the class name and again afterward. I moved that up to here https://github.com/HubSpot/cms-theme-boilerplate/pull/500/files#diff-6ee708090f4d589c0bdd00a3d14ce0ea48a64c24f1a032e9a90500466d26d7deR110

Comment (jasonnrosa) [reply src/modules/menu.module/module.html:115]: Ah okay got it - makes sense! Sorry - I had missed that line when reviewing the first time around

**Review by jasonnrosa — APPROVED** [review]

**Pull request merged**